runtime/src/consensus/state/staking.rs

```diff
@@ -0,0 +1,72 @@
+use std::collections::BTreeMap;
+
+use anyhow::{anyhow, Result};
+use io_context::Context;
+
+use crate::{
+    common::{
+        cbor,
+        key_format::{KeyFormat, KeyFormatAtom},
+    },
+    consensus::{
+        address::Address,
+        staking::{Account, Delegation},
+    },
+    key_format,
+    storage::mkvs::ImmutableMKVS,
+};
+
+/// Consensus staking state wrapper.
+pub struct ImmutableState<'a, T: ImmutableMKVS> {
+    pub mkvs: &'a T,
+}
+
+key_format!(AccountsKeyFmt, Address, 0x50);
+key_format!(TotalSupplyKeyFmt, (), 0x51);
+key_format!(CommonPoolKeyFmt, (), 0x52);
+key_format!(DelegationKeyFmt, (Address, Address), 0x53);
+key_format!(DebondingDelegationKeyFmt, (Address, Address, u64), 0x54);
+key_format!(DebondingQueueKeyFmt, (u64, Address, Address), 0x55);
+key_format!(ParametersKeyFmt, (), 0x56);
+key_format!(LastBlockFees, (), 0x57);
+key_format!(EpochSigningKeyFmt, (), 0x58);
+key_format!(GovernanceDepositsKeyFmt, (), 0x59);
+
+// TODO: implement the rest. Add some tests?
+impl<'a, T: ImmutableMKVS> ImmutableState<'a, T> {
+    pub fn account(self, ctx: Context, address: Address) -> Result<Account> {
```

Comment:
Why do you consume `self` in these methods?

Reply:
Yeah shouldn't. Will update